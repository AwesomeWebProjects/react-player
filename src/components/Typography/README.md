Typograpy

Generating the tag `element` by default:

```js
<Typography variant="displayXXL">Display XXL</Typography>
<Typography variant="displayXL">Display XL</Typography>
<Typography variant="displayL">Display L</Typography>
<Typography variant="displayM">Display M</Typography>
<Typography variant="displayS">Display S</Typography>
<Typography variant="subtitle">Subtitle</Typography>
```

Typograpy custom:

Setting the `element` as a tag and keeping styles as the `variant` property:

```js
<Typography element="p" variant="buttonM">Button M</Typography>
<Typography element="p" variant="bodyM">Body M</Typography>
<Typography element="p" variant="buttonS">Button S</Typography>
<Typography element="p" variant="bodyS">Body S</Typography>
<Typography element="p" variant="cardContent">CardContent</Typography>
<Typography element="p" variant="buttonXS">Button XS</Typography>
<Typography element="p" variant="bodyXS">Body XS</Typography>
<Typography element="p" variant="buttonXXS">Button XXS</Typography>
```
