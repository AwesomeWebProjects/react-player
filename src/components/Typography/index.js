import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import style from './style.module.css'

const Typography = ({ className, children, innerHTML, variant, element }) => {
  const typographyClassNames = classNames(style.typography, className, style[variant])

  const elementMap = {
    displayXXL: 'h1',
    displayXL: 'h2',
    displayL: 'h3',
    displayM: 'h4',
    displayS: 'h5',
    subtitle: 'h6',
    bodyL: 'p',
    buttonM: 'span',
    bodyM: 'p',
    buttonS: 'span',
    bodyS: 'p',
    cardContent: 'p',
    buttonXS: 'span',
    bodyXS: 'p',
    buttonXXS: 'p',
  }

  const TypographyElement = element || elementMap[variant]

  if (innerHTML) {
    return (
      <TypographyElement
        className={typographyClassNames}
        dangerouslySetInnerHTML={{ __html: innerHTML }}
      />
    )
  }

  return <TypographyElement className={typographyClassNames}>{children}</TypographyElement>
}

Typography.propTypes = {
  className: PropTypes.string,
  element: PropTypes.string,
  variant: PropTypes.oneOf([
    'displayXXL',
    'displayXL',
    'displayL',
    'displayM',
    'displayS',
    'subtitle',
    'bodyL',
    'buttonM',
    'bodyM',
    'buttonS',
    'bodyS',
    'cardContent',
    'buttonXS',
    'bodyXS',
    'buttonXXS',
  ]),
  innerHTML: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
    PropTypes.object,
  ]),
}

Typography.defaultProps = {
  className: null,
  element: null,
  innerHTML: null,
  children: null,
  variant: 'bodyS',
}

export default Typography
