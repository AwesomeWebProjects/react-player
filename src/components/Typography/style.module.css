.typography {
  line-height: 130%;
  margin-top: 0;
  font-family: 'Inter', sans-serif;
}

.displayXXL {
  font-size: var(--begrowth-font-size-xxxl);
  font-weight: var(--begrowth-font-weight-black);
}

.displayXL {
  font-size: var(--begrowth-font-size-xxl);
  font-weight: var(--begrowth-font-weight-black);
}

.displayL {
  font-size: var(--begrowth-font-size-xl);
  font-weight: var(--begrowth-font-weight-black);
}

.displayM {
  font-size: var(--begrowth-font-size-l);
  font-weight: var(--begrowth-font-weight-black);
}

.displayS {
  font-size: var(--begrowth-font-size-s);
  font-weight: var(--begrowth-font-weight-black);
}

.subtitle {
  font-size: var(--begrowth-font-size-s);
  font-weight: var(--begrowth-font-weight-medium);
}

.bodyL {
  font-size: var(--begrowth-font-size-l);
  font-weight: var(--begrowth-font-weight-regular);
}

.buttonM {
  font-size: var(--begrowth-font-size-m);
  font-weight: var(--begrowth-font-weight-medium);
}

.bodyM {
  font-size: var(--begrowth-font-size-m);
  font-weight: var(--begrowth-font-weight-regular);
}

.buttonS {
  font-size: var(--begrowth-font-size-s);
  font-weight: var(--begrowth-font-weight-regular);
}

.bodyS {
  font-size: var(--begrowth-font-size-s);
  font-weight: var(--begrowth-font-weight-regular);
}

.cardContent {
  font-size: var(--begrowth-font-size-xs);
  font-weight: var(--begrowth-font-weight-bold);
}

.buttonXS {
  font-size: var(--begrowth-font-size-xs);
  font-weight: var(--begrowth-font-weight-medium);
}

.bodyXS {
  font-size: var(--begrowth-font-size-xs);
  font-weight: var(--begrowth-font-weight-regular);
}

.buttonXXS {
  font-size: var(--begrowth-font-size-xxs);
  font-weight: var(--begrowth-font-weight-medium);
}
