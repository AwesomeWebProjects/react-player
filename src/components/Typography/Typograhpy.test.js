/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import Typography from './index'

test('Typograpy: simple render test', () => {
  const component = renderer.create(<Typography>Hello worldddd</Typography>)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
