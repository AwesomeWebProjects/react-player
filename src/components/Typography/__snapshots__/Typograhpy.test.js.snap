// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typograpy: simple render test 1`] = `
<p
  className="typography bodyS"
>
  Hello worldddd
</p>
`;
